src/sentry/sentry_metrics/kafka_metrics_interface.py

```diff
@@ -0,0 +1,137 @@
+from concurrent.futures import Future
+from datetime import datetime
+from typing import Mapping, Optional, Sequence, Union
+
+from arroyo import Topic
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer, build_kafka_configuration
+from arroyo.types import BrokerValue
+from django.conf import settings
+
+from sentry.sentry_metrics.metrics_interface import GenericMetricsBackend
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.utils import json
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options
+
+
+def build_mri(metric_name: str, type: str, use_case_id: UseCaseID, unit: Optional[str]) -> str:
+    mri_unit = "none" if unit is None else unit
+    return f"{type}:{use_case_id.value}/{metric_name}@{mri_unit}"
+
+
+# def set_future(f: Future[BrokerValue[KafkaPayload]]) -> Union[Future[None], Future[Any]]:
+#     new_future: Union[Future[None], Future[Any]] = Future()
+
+#     if f.exception() is not None:
+#         new_future.set_exception(f.exception())
+#     else:
+#         new_future.set_result(f.result)
+
+#     return new_future
+
+
+class KafkaMetricsBackend(GenericMetricsBackend):
```

Comment:
nit: let's call this file kafka_metrics_backend.py (the interface is defined in metrics_interface.py)

Reply:
I suggest you follow the same naming convention as the indexer (which is used by quite a number of modules in Sentry):
the abstract backend is in a file called `base.py` and  the kafka backend is named `kafka`.